Signal/src/AppDelegate.m

```diff
@@ -239,7 +239,11 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
                                                [TSSocketManager becomeActiveFromForeground];
                                                [[Environment getCurrent].contactsManager verifyABPermission];
                                            }];
-
+    
+    if ([TSAccountManager isRegistered]) {
+        [[PushManager sharedManager] applicationDidBecomeActive];
```

Comment:
> I've gotten into the practice of putting things that read from the DB into an async block...

I'm concerned about thready safety, since this moves all of the logic in that completion block off the main thread.  Is all of the logic in that completion thread-safe?  Is any of it expensive? Or is it calling `[TSAccountManager ifRegistered]` that is expensive?